### PR TITLE
add okd images for sriov operator

### PR DESCRIPTION
**Pull request by SchSeba**

### CODE REVIEW

**Comment by SchSeba** [comment]
/pj-rehearse pull-ci-openshift-sriov-network-operator-master-security

**Comment by SchSeba** [comment]
/pj-rehearse pull-ci-openshift-sriov-network-operator-master_okd-scos-images

**Comment by SchSeba** [comment]
/pj-rehearse pull-ci-openshift-sriov-network-operator-master_okd-scos-images

File: ci-operator/jobs/openshift/sriov-network-operator/openshift-sriov-network-operator-master_okd-scos-presubmits.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+presubmits:
+  openshift/sriov-network-operator:
+  - agent: kubernetes
+    always_run: true
+    branches:
+    - ^master_okd-scos$
+    - ^master_okd-scos-
+    cluster: build03
+    context: ci/prow/images
```
Comment (Prashanth684) [inline ci-operator/jobs/openshift/sriov-network-operator/openshift-sriov-network-operator-master_okd-scos-presubmits.yaml:9]: this needs a different name like `okd-scos-images`

File: ci-operator/jobs/openshift/sriov-network-operator/openshift-sriov-network-operator-master_okd-scos-presubmits.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+presubmits:
+  openshift/sriov-network-operator:
+  - agent: kubernetes
+    always_run: true
```
Comment (Prashanth684) [inline ci-operator/jobs/openshift/sriov-network-operator/openshift-sriov-network-operator-master_okd-scos-presubmits.yaml:4]: you want this to be false and `optional: true`

File: ci-operator/jobs/openshift/sriov-network-operator/openshift-sriov-network-operator-master_okd-scos-presubmits.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    - ^master_okd-scos$
+    - ^master_okd-scos-
+    cluster: build03
+    context: ci/prow/images
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+    labels:
+      ci.openshift.io/generator: prowgen
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
```
Comment (Prashanth684) [inline ci-operator/jobs/openshift/sriov-network-operator/openshift-sriov-network-operator-master_okd-scos-presubmits.yaml:15]: need these two labels:
```
 ci-operator.openshift.io/is-promotion: "true"
 ci-operator.openshift.io/variant: okd-scos
```

File: ci-operator/jobs/openshift/sriov-network-operator/openshift-sriov-network-operator-master_okd-scos-presubmits.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
+presubmits:
+  openshift/sriov-network-operator:
+  - agent: kubernetes
+    always_run: true
+    branches:
+    - ^master_okd-scos$
```
Comment (Prashanth684) [inline ci-operator/jobs/openshift/sriov-network-operator/openshift-sriov-network-operator-master_okd-scos-presubmits.yaml:6]: do you actually have a branch `master-okd-scos` in the repo? i think this should just be `master`

File: ci-operator/jobs/openshift/sriov-network-operator/openshift-sriov-network-operator-master_okd-scos-presubmits.yaml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    name: pull-ci-openshift-sriov-network-operator-master_okd-scos-images
+    rerun_command: /test images
+    spec:
+      containers:
+      - args:
+        - --gcs-upload-secret=/secrets/gcs/service-account.json
+        - --image-import-pull-secret=/etc/pull-secret/.dockerconfigjson
+        - --report-credentials-file=/etc/report/credentials
+        - --target=[images]
+        - --target=[release:latest]
```
Comment (Prashanth684) [inline ci-operator/jobs/openshift/sriov-network-operator/openshift-sriov-network-operator-master_okd-scos-presubmits.yaml:25]: needs `--variant=okd-scos`

**Comment by Prashanth684** [comment]
@SchSeba i would look at this PR: https://github.com/openshift/release/pull/50439 to see if you have the proper labels/ flags for the job config to work correctly

**Comment by SchSeba** [comment]
/pj-rehearse pull-ci-openshift-sriov-network-operator-master-okd-scos-images

File: ci-operator/jobs/openshift/sriov-network-operator/openshift-sriov-network-operator-master_okd-scos-presubmits.yaml (hunk 2 above)
Comment (SchSeba) [reply ci-operator/jobs/openshift/sriov-network-operator/openshift-sriov-network-operator-master_okd-scos-presubmits.yaml]: I can see in the multus example you point out it's the same
https://github.com/openshift/release/pull/50439

and I want this one to run on every merge to master

File: core-services/image-mirroring/openshift/mapping_origin_4_16 [hunk 6]
```diff
@@ -13,4 +13,7 @@ registry.ci.openshift.org/origin/4.16:okd-rpms quay.io/openshift/origin-okd-rpms
 registry.ci.openshift.org/origin/4.16:openstack-installer quay.io/openshift/origin-openstack-installer:4.16 quay.io/openshift/origin-openstack-installer:4.16.0 quay.io/openshift/origin-openstack-installer:latest
 registry.ci.openshift.org/origin/4.16:operator-marketplace quay.io/openshift/origin-operator-marketplace:4.16 quay.io/openshift/origin-operator-marketplace:4.16.0 quay.io/openshift/origin-operator-marketplace:latest
 registry.ci.openshift.org/origin/4.16:ovirt-installer quay.io/openshift/origin-ovirt-installer:4.16 quay.io/openshift/origin-ovirt-installer:4.16.0 quay.io/openshift/origin-ovirt-installer:latest
+registry.ci.openshift.org/origin/4.16:sriov-network-config-daemon quay.io/openshift/origin-sriov-network-config-daemon:4.16 quay.io/openshift/origin-sriov-network-config-daemon:4.16.0 quay.io/openshift/origin-sriov-network-config-daemon:latest
```
Comment (Prashanth684) [inline core-services/image-mirroring/openshift/mapping_origin_4_16:16]: @jupierce will we see these images mirrored after the postsubmit jobs are executed?

File: ci-operator/config/openshift/sriov-network-operator/openshift-sriov-network-operator-master__okd-scos.yaml [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+build_root:
+  image_stream_tag:
+    name: builder
+    namespace: ocp
+    tag: rhel-8-golang-1.21-openshift-4.16
+images:
+- dockerfile_path: Dockerfile.rhel7
+  from: origin_scos-4.15_base-stream9
+  inputs:
+    ocp_4.16_base-rhel9:
```
Comment (jupierce) [inline ci-operator/config/openshift/sriov-network-operator/openshift-sriov-network-operator-master__okd-scos.yaml:23]: I would not expect replacements with RHEL based images to be included.

**Comment by SchSeba** [comment]
/pj-rehearse pull-ci-openshift-sriov-network-operator-master-okd-scos-images

**Comment by SchSeba** [comment]
great this one works

```
cat /etc/os-release 
NAME="CentOS Stream"
VERSION="9"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="9"
PLATFORM_ID="platform:el9"
PRETTY_NAME="CentOS Stream 9"
ANSI_COLOR="0;31"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:centos:centos:9"
HOME_URL="https://centos.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux 9"
REDHAT_SUPPORT_PRODUCT_VERSION="CentOS Stream"

```

**Pull request merged**